crates/ruma-common/src/events/image.rs

```diff
@@ -0,0 +1,333 @@
+        self.captions()
+            .iter()
+            .find(|content| content.mimetype == "text/plain")
+            .map(|content| content.body.as_ref())
+    }
+
+    /// Get the HTML representation of this caption.
+    pub fn find_html(&self) -> Option<&str> {
+        self.captions()
+            .iter()
+            .find(|content| content.mimetype == "text/html")
+            .map(|content| content.body.as_ref())
+    }
+}
+
+/// Image content.
+#[derive(Default, Clone, Debug, Serialize, Deserialize)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+pub struct ImageContent {
+    /// The height of the image in pixels.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub height: Option<UInt>,
+
+    /// The width of the image in pixels.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub width: Option<UInt>,
+}
+
+impl ImageContent {
+    /// Creates a new empty `ImageContent`.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Creates a new `ImageContent` with the given width and height.
+    pub fn with_size(width: UInt, height: UInt) -> Self {
+        Self { height: Some(height), width: Some(width) }
+    }
+}
+
+/// The payload for an extensible image message.
+#[derive(Clone, Debug, Serialize, Deserialize, EventContent)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[ruma_event(type = "m.image", kind = MessageLike)]
+pub struct ImageEventContent {
+    /// The text representation of the message.
+    #[serde(flatten)]
+    pub message: MessageContent,
+
+    /// The file content of the message.
+    #[serde(rename = "org.matrix.msc1767.file")]
+    pub file: FileContent,
+
+    /// The image content of the message.
+    #[serde(rename = "org.matrix.msc1767.image")]
+    pub image: Box<ImageContent>,
+
+    /// The thumbnails of the message.
+    #[serde(
+        rename = "org.matrix.msc1767.thumbnail",
+        default,
+        skip_serializing_if = "Thumbnails::is_empty"
+    )]
+    pub thumbnail: Thumbnails,
+
+    /// The thumbnails of the message.
+    #[serde(
+        rename = "org.matrix.msc1767.caption",
+        default,
+        skip_serializing_if = "Captions::is_empty"
+    )]
+    pub caption: Captions,
+
+    /// Information about related messages for [rich replies].
+    ///
+    /// [rich replies]: https://spec.matrix.org/v1.2/client-server-api/#rich-replies
+    #[serde(flatten, skip_serializing_if = "Option::is_none")]
+    pub relates_to: Option<Relation>,
+}
+
+impl ImageEventContent {
+    /// Creates a new non-encrypted `ImageEventContent` with the given plain text message, url,
+    /// file info, image info, thumbnail and caption.
+    pub fn plain(
+        message: impl Into<String>,
+        url: Box<MxcUri>,
+        file_info: Option<Box<FileContentInfo>>,
+        image_info: Option<Box<ImageContent>>,
+        thumbnail: Option<Thumbnails>,
+        caption: Option<Captions>,
```

Comment:
Please remove all of these optional parameters. They make initialization look rather ugly. Users can assign the fields that have a default after construction.